File: q3354/q3354.py
class Solution:
    def countValidSelections(self, nums: List[int]) -> int:
        total = 0
        numsSum = sum(nums)
        for i, num in enumerate(nums):
            if num == 0:
                if self.explore(nums[:], i, 1, numsSum): total += 1
                if self.explore(nums[:], i, -1, numsSum): total += 1
        return total
    
    def explore(self, nums, currIdx, direction, numsSum) -> bool:
        while currIdx >= 0 and currIdx < len(nums):
            if nums[currIdx] > 0:
                # decrement the currIdx value by 1
                nums[currIdx] -= 1
                # decrement tht total sum by 1
                numsSum -= 1

                # reverse direction
                direction = -1 * direction
            
            # take a step in the direction
            currIdx = currIdx + direction

        # only need to check the sum of all elements
        return numsSum == 0

File: q3354/q3354.java
class Solution {
    public int countValidSelections(int[] nums) {
        int total = 0;
        int numsSum = 0;
        for (int num : nums) numsSum += num;

        for (int i = 0; i < nums.length; i++) {
            if (nums[i] == 0) {
                if (explore(nums.clone(), i, 1, numsSum)) total++;
                if (explore(nums.clone(), i, -1, numsSum)) total++;
            }
        }
        return total;
    }

    private boolean explore(int[] nums, int currIdx, int direction, int numsSum) {
        while (currIdx >= 0 && currIdx < nums.length) {
            if (nums[currIdx] > 0) {
                nums[currIdx]--;
                numsSum--;
                direction = -1 * direction;
            }
            currIdx = currIdx + direction;
        }
        return numsSum == 0;
    }
}
